redisvl/extensions/llmcache/semantic.py

```diff
@@ -101,25 +81,42 @@ def __init__(
                 model="sentence-transformers/all-mpnet-base-v2"
             )
 
-        schema = SemanticCacheIndexSchema.from_params(name, vectorizer.dims)
+        # Create semantic cache schema
+        schema = SemanticCacheIndexSchema.from_params(name, prefix, vectorizer.dims)
+
+        # Process fields
+        self.return_fields = [
+            self.entry_id_field_name,
+            self.prompt_field_name,
+            self.response_field_name,
+            self.inserted_at_field_name,
+            self.updated_at_field_name,
+            self.metadata_field_name,
+        ]
+
+        if filterable_fields is not None:
+            for filter_field in filterable_fields:
+                if (
+                    filter_field["name"] in self.return_fields
+                    or filter_field["name"] == "key"
+                ):
+                    raise ValueError(
+                        f'{filter_field["name"]} is a reserved field name for the semantic cache schema'
+                    )
+                schema.add_field(filter_field)
```

Comment:
I really like the idea of allowing fully customizable fields for people to add and filter on!